presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Partition.java

```diff
@@ -50,6 +51,7 @@ public class Partition
 
     @JsonCreator
     public Partition(
+            @JsonProperty("catalogName") Optional<String> catalogName,
```

Comment:
Consider adding a constructor which maintains all the original parameters (without the Json annotations) in order to reduce changing call sites all over the codebase